Library/Homebrew/api/cask.rb

```diff
@@ -20,16 +20,34 @@ def fetch_source(token, git_head: nil)
           Homebrew::API.fetch_homebrew_cask_source token, git_head: git_head
         end
 
+        sig { returns(T::Boolean) }
+        def prepare_json_data
+          json_casks, updated = Homebrew::API.fetch_json_api_file "cask.json",
+                                                                  target: HOMEBREW_CACHE_API/"cask.json"
+
+          @all_casks = json_casks.to_h do |json_cask|
+            [json_cask["token"], json_cask.except("token")]
+          end
+
+          updated
+        end
+        private :prepare_json_data
+
         sig { returns(Hash) }
         def all_casks
-          @all_casks ||= begin
-            json_casks = Homebrew::API.fetch_json_api_file "cask.json",
-                                                           target: HOMEBREW_CACHE_API/"cask.json"
-
-            json_casks.to_h do |json_cask|
-              [json_cask["token"], json_cask.except("token")]
-            end
+          unless defined?(@all_casks)
```

Comment:
Feels a bit weird to set this elsewhere in another function. Would be nice to still cache it in this method and have `prepare_json_data` cache something else.

Reply:
There's a bit of a loop problem where `write_names` needs `@all_casks`, but `all_casks` function calls `write_names`. I wasn't sure how else to solve that.

Reply:
@Bo98 sounds like both functions should call another function that does the downloading and caching. Alternatively, have a single function named `download_and_cache_all_casks!` or something that replaces both

Reply:
That's exactly what I was trying to do here. So is it just a case of renaming `prepare_json_data` to `download_and_cache_all_formulae!`?

Reply:
@Bo98 Yes and pass values through return values rather than rely on an instance variable being set/defined in another method.

Reply:
We however have two instance variables to set but only want to fetch/read once. So if `download_and_cache_all_formulae!` can't cache it in `@all_formulae/@all_aliases` then where should it cache it instead?

Reply:
Either return them both, memoise them in separate methods which each return the cached value or memoize a hash instead with both values.

Reply:
It needs to return a boolean information from the fetch too, but I saw the existing Cachable model we have and thought that could apply here as that gets rid of instance variables entirely.